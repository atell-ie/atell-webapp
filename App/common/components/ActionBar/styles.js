import { variables } from "../../styles";

const { actionBarMinHeight } = variables;

export default (theme) => ({
    appBar: {
        background: theme.palette.primary.lightBackground,
        borderTop: `1px solid ${theme.palette.primary.lightBorder}`,
        minHeight: `${actionBarMinHeight}px`,
        position: "fixed",
        display: "flex",
        justifyContent: "center",
        top: "auto",
        bottom: 0
    }
});
