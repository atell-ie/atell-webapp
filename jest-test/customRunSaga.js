import { runSaga } from "redux-saga";

const customRunSaga = async (saga, initAction) => {
    const dispatched = [];
    await runSaga(
        {
            dispatch: (action) => dispatched.push(action),
        },
        saga,
        initAction
    ).toPromise();

    return dispatched;
};

export default customRunSaga;
