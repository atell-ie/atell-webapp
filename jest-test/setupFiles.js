import "@testing-library/jest-dom";
import "jest-canvas-mock";
const imageCompression = require("browser-image-compression");
const image = require("@/jest-test-data/Camera/image");
const dotenv = require("dotenv");
dotenv.config({ path: "./App/config/.env" });

process.env.HOST = "local";

class LocalStorageMock {
    constructor() {
        this.store = {};
    }
    clear() {
        this.store = {};
    }
    getItem(key) {
        return this.store[key] || null;
    }
    setItem(key, value) {
        this.store[key] = String(value);
    }
    removeItem(key) {
        delete this.store[key];
    }
}

global.localStorage = new LocalStorageMock();

// mocking image compress module
async function mockedCompressedFunc() {
    return {};
}
mockedCompressedFunc.getFilefromDataUrl = () => jest.fn();
mockedCompressedFunc.getDataUrlFromFile = () => image;

// need to mock compression, otherwise test failing
jest.mock("browser-image-compression", () => ({
    __esModule: true, // this property makes it work
    default: mockedCompressedFunc,
}));
