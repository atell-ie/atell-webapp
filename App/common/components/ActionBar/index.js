import React from "react";
import { AppBar, Toolbar } from "@mui/material";
import styles from "./styles";

const ActionBar = ({ children }) => {
    return (
        <AppBar position="sticky" sx={styles.appBar} elevation={0}>
            <Toolbar>{children}</Toolbar>
        </AppBar>
    );
};

export default ActionBar;
