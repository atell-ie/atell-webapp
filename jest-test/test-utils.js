import React from "react";

import { render as rtlRender } from "@testing-library/react";
import { ThemeProvider, StyledEngineProvider } from "@mui/material/styles";
import LocalizationProvider from "@mui/lab/LocalizationProvider";
import AdapterDateFns from "@mui/lab/AdapterDateFns";
import MomentUtils from "@date-io/moment";
import { theme } from "../App/common/styles";
import { I18nextProvider } from "react-i18next";
import i18n from "../App/common/i18n";

import { createStore } from "redux";
import { Provider } from "react-redux";

import rootReducer from "../App/Store/root-reducer";

const customRender = (ui, { initialState, ...renderOptions } = {}) => {
    const store = createStore(rootReducer);

    const Wrapper = ({ children }) => {
        return (
            <LocalizationProvider
                utils={MomentUtils}
                dateAdapter={AdapterDateFns}
            >
                <StyledEngineProvider injectFirst>
                    <ThemeProvider theme={theme}>
                        <I18nextProvider i18n={i18n}>
                            <Provider store={store}>{children}</Provider>
                        </I18nextProvider>
                    </ThemeProvider>
                </StyledEngineProvider>
            </LocalizationProvider>
        );
    };

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender };
