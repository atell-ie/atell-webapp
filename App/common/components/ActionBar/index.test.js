import React from "react";
import { customRender, waitFor } from "@/jest-test/test-utils";
import ActionBar from ".";

describe("ActionBar", () => {
    test("default behaviour ", async () => {
        const { container, getByText } = customRender(
            <ActionBar>
                <div>test</div>
            </ActionBar>
        );

        await waitFor(() => {
            const startAudit = "test";
            expect(getByText(startAudit)).toBeInTheDocument();
            expect(container).toMatchSnapshot();
        });
    });
});
