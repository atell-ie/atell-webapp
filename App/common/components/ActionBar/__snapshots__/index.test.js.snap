// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar default behaviour  1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky makeStyles-appBar-1 css-y0narc-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div>
        test
      </div>
    </div>
  </header>
</div>
`;
